File: Computabilidade-e-Complexidade-de-Algoritmos-main/python/11.py
def afd_impar_as(s):
    estado = 'q0'
    for char in s:
        if estado == 'q0' and char == 'a':
            estado = 'q1'
        elif estado == 'q1' and char == 'a':
            estado = 'q0'
    return estado == 'q1'

# Testes
print(afd_impar_as("a"))    # True
print(afd_impar_as("aa"))   # False
print(afd_impar_as("aba"))  # True

File: Computabilidade-e-Complexidade-de-Algoritmos-main/java/11.java
public class AFDImparAs {
    public static boolean imparAs(String s) {
        String estado = "q0";
        for (char c : s.toCharArray()) {
            if (estado.equals("q0") && c == 'a') {
                estado = "q1";
            } else if (estado.equals("q1") && c == 'a') {
                estado = "q0";
            }
        }
        return estado.equals("q1");
    }

    public static void main(String[] args) {
        System.out.println(imparAs("a"));    // True
        System.out.println(imparAs("aa"));   // False
        System.out.println(imparAs("aba"));  // True
    }
}
